Converts Results SFC into a view

matt-bernhardt opened the pull request:
This moves the search results from `App.vue` into their own view, along with the Facets and Pagination components. This simplifies some of the page rendering at the App level.

#### What are the relevant tickets?
- https://mitlibraries.atlassian.net/browse/DISCO-144

#### Includes new or updated dependencies?
NO

JPrevost commented on src/App.vue:
```diff
@@ -5,43 +5,27 @@
             v-model:status="status"
             v-model:hits="hits"
           />
+          <div v-if="status.loading" class="panel panel-info">
```
We have very similar blocks in two router views now. I want to consider moving this to a Component in the future (just thinking out loud not suggesting a change here)

JPrevost replied:
I think this block needs to be moved into the router view. You can see this not populating by searching for something like `asdfasdfasdf`

JPrevost commented on src/App.vue:
```diff
@@ -5,43 +5,27 @@
-              v-bind:key="result.id"
-            />
-            <Pagination v-if="showPagination" msg="Pagination bar" />
+            <router-view @results="receiveSummary" />
```
Is this the same `<router-view>` tag used by all routes?

matt-bernhardt replied:
Yeah, this is the one slot in the template where any route would display (so the results and record views currently)

JPrevost commented on src/views/Results.vue:
```diff
@@ -0,0 +1,105 @@
+            (this.results = response.data.results),
+            (this.hits = response.data.hits),
+            (this.status.ready = true),
+            this.$emit("results", { hits: this.hits, query: this.query })
```
Oh hey, our first `emit`!

JPrevost replied:
Although now I'm thinking about it maybe we could move the search summary to this component as it's only used here and it would simplify a fair amount. Hmmm. Something to consider for later (but not now).

JPrevost commented on src/views/Results.vue:
```diff
@@ -0,0 +1,105 @@
+        error_message: "",
+        errored: false,
+        loading: false,
+        ready: false
```
I don't see where `ready` is used for anything.

matt-bernhardt replied:
The `ready` code here may not be used - I'll take it out.

JPrevost reviewed (changes requested):
I think the easiest thing to do for now is to move the error display to the `Results.vue` template. I'm not sure I want to do this longterm but it would probably be easiest to land that way and then decide later how to normalize error handling across all of our routed views.

I'm not sure we gain much by having a separate `SearchMetadata` component and it does add complexity in both `App.vue` and `Results.vue` to pass the data around. I'm not against keeping it as is, but take a look at it from this perspective and see if you agree it might be better all in `Results.vue`. I'm totally cool with whatever you decide for this.

matt-bernhardt commented:
Ok, coming back to this after lunch and I think I've addressed your concerns @JPrevost - the two status blocks have been moved down to Results.vue, and I've removed the un-rendered `status.ready` value. Ultimately I think you're right, it might be good to have a common component that handles all of this - but I'm not sure that this is the PR for that amount of work.

JPrevost reviewed (approved):
👍🏻
